third-party/realdds/src/topics/imu-msg.cpp

```diff
@@ -0,0 +1,116 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include <realdds/topics/ros2/ros2imu.h>
+#include <realdds/topics/imu-msg.h>
+#include <realdds/topics/ros2/ros2imuPubSubTypes.h>
+
+#include <realdds/dds-topic.h>
+#include <realdds/dds-topic-reader.h>
+#include <realdds/dds-topic-writer.h>
+#include <realdds/dds-utilities.h>
+#include <realdds/dds-time.h>
+
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+
+
+namespace realdds {
+namespace topics {
+
+
+imu_msg::imu_msg()
+{
+    // Everything (Vector3, Quaternion) should be zero'ed out except the arrays
+    // "If you have no estimate for one of the data elements (e.g. your IMU doesn't produce an orientation estimate),
+    // please set element 0 of the associated covariance matrix to -1"
+    imu_data().orientation_covariance() = { -1., 0., 0., 0., 0., 0., 0., 0., 0. };
+    // "you can initiate the sensor covariance matrix by giving some small values on the diagonal"
+    double const cov = 0.01; // TODO from context?
+    imu_data().linear_acceleration_covariance() = { cov, 0., 0., 0., cov, 0., 0., 0., cov };
```

Comment:
Shouldn't the first element be set to -1 (as in orientation?)

Reply:
We do have linear acceleration (accel) and angular velocity (gyro).
This is a copy of the code from the ROS wrapper.